core/src/layers/prometheus.rs

```diff
@@ -93,32 +89,117 @@ use crate::*;
 ///     println!("## Prometheus Metrics");
 ///     println!("{}", String::from_utf8(buffer.clone()).unwrap());
 ///     Ok(())
-/// }
+/// # }
 /// ```
 #[derive(Clone, Debug)]
 pub struct PrometheusLayer {
-    registry: Registry,
+    interceptor: PrometheusInterceptor,
+}
+
+impl Default for PrometheusLayer {
```

Comment:
How about removing `Default` for `PrometheusLayer`? We can expect all users build layers by our `builder` directly?